api/src/main/java/org/eclipse/microprofile/config/Config.java

```diff
@@ -181,6 +181,49 @@
         return getOptionalValue(propertyName, arrayType).map(Arrays::asList);
     }
 
+    /**
+     * Return the resolved configuration properties type with the specified prefix. 
+     * The type declaration can be annotated with 
+     * {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties}. 
+     * If the type is annotated with {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties}, 
+     * the prefix supplied in this method overrides the prefix associated with the annotation 
+     * {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties}.
+     * 
+     * @param <T> 
+     *              The Class Type
+     * @param configProperties
+     *              The class that contains a number of fields that maps to corresponding configuration properties. 
+     *              The type declaration can be annotated with 
+     *              {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties}.
+     *              The prefix as the method parameter overrides the prefix set on the type.                
+     *              This class must contain a zero-arg constructor.
+     * @param prefix
+     *              The prefix for the configuration properties declared on the class configProperties
```

Comment:
Can it be `null`?  What happens if it is `""`?  (And so on.)

Reply:
We should probably throw a `IllegalArgumentException` is the value is null, since you have the method without the `prefix`. If you pass an empty String, there is no `prefix` involved, so the properties are looked up as is.

Reply:
> Can it be null? What happens if it is ""? (And so on.)

It can be "", which is used to override the prefix on the annotation. Passing in null is the same as calling getConfigProperties(Class), which means no prefix is passed in.  I disagree to throw IllegalArgumentException as in the implemenation of getConfigProperties(Class) is to call into getConfigProperties(Class, null). I can clarify this in the javadoc.

Reply:
Surely all of this discussion should be in the javadoc.

Reply:
I'll clarify this in the javadoc.

Reply:
updated